src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -683,7 +689,7 @@ bool fx_muxer_t::resolve_sdk_dotnet_path(const pal::string_t& own_dir, const pal
                 }
             }
         }
-        if (retval.empty())
+        if (retval.empty() && !cli_version_specified)
```

Comment:
This fixes the repro steps I had listed in #2336, but we also need to make sure that when we pick the latest version when there's no global.json, it's the latest across all hive_dirs here. 

You can leave #2336 open and tackle that part separately if you want.